apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -458,9 +459,32 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
   end
 
   def set_device_connected!(db_client, device_id, timestamp_ms, ip_address) do
+    set_connection_info!(db_client, device_id, timestamp_ms, ip_address)
+
+    ttl = heartbeat_interval_seconds() * 8
+    refresh_device_connected!(db_client, device_id, ttl)
+  end
+
+  def maybe_refresh_device_connected!(db_client, device_id) do
+    with {:ok, remaining_ttl} <- get_connected_remaining_ttl(db_client, device_id) do
+      if remaining_ttl < heartbeat_interval_seconds() * 2 do
+        Logger.info("Refreshing connected status for device #{inspect(device_id)}")
```

Comment:
I would use Logger.debug here

Reply:
anyway beware that we don't need anymore to explicitly log device_id and realm, those information are attached to the processes.

Reply:
we also add a tag to every kind of message, to make them machine parsable.